### PR TITLE
Fixing package building for Arch that was broken a few versions ago

### PR DESCRIPTION
**Pull request by avielw**
Since a few versions ago, wazuh-packages is unable to create packages for Arch Linux. This commit fixes that.
(This is related to issue: #1649)

### CODE REVIEW

File: arch/SPECS/wazuh-agent/arch/PKGBUILD [hunk 1]
```diff
@@ -53,6 +53,7 @@ package() {
 	# Copying systemd file
 	mkdir -p ${pkgdir}/usr/lib/systemd/system/
 	install -m 0644 src/init/templates/wazuh-agent.service ${pkgdir}/usr/lib/systemd/system/
+    sed -i 's:WAZUH_HOME_TMP:PARAM_INSTALLATION_DIR:g' ${pkgdir}/usr/lib/systemd/system/wazuh-agent.service
```
Comment (rauldpm) [inline arch/SPECS/wazuh-agent/arch/PKGBUILD:56]: Please, fix file tabulation https://github.com/wazuh/wazuh-packages/blob/f79bec455fff232157f567bdd922dac8ad32ab62/arch/SPECS/wazuh-agent/arch/PKGBUILD

File: arch/SPECS/wazuh-agent/arch/wazuh.install [hunk 2]
```diff
@@ -1,8 +1,8 @@
 DIR="PARAM_INSTALLATION_DIR"
 SCRIPTS_DIR="PARAM_INSTALLATION_SCRIPTS_DIR"
 SCA_BASE_DIR="${SCRIPTS_DIR}/sca"
-USER="ossec"
-GROUP="ossec"
+USER="wazuh"
+GROUP="wazuh"
```
Comment (rauldpm) [inline arch/SPECS/wazuh-agent/arch/wazuh.install:5]: Does this package remove the ossec user and group in an upgrade process?

You could add installation, uninstallation and upgrade tests to the PR comment to give more visibility on the functionality of the package, you can put the output and the tests that you consider appropriate to show us in dropdowns through 


```
<details>\<summary>title\</summary>

output

</details>
```

Would be see as

<details><summary>title</summary>

output

</details>

File: arch/SPECS/wazuh-agent/arch/wazuh.install [hunk 3]
```diff
@@ -42,7 +42,7 @@ common() {
     fi
 
     touch ${DIR}/logs/active-responses.log
-    chown ossec:ossec ${DIR}/logs/active-responses.log
+    chown wazuh:wazuh ${DIR}/logs/active-responses.log
```
Comment (rauldpm) [inline arch/SPECS/wazuh-agent/arch/wazuh.install:45]: Use `${USER}` and `${GROUP}` variables

File: arch/SPECS/wazuh-agent/arch/wazuh.install [hunk 4]
```diff
@@ -55,7 +55,7 @@ common() {
 
     # Fix /etc/ossec-init.conf
     chmod 640 /etc/ossec-init.conf
-    chown root:ossec /etc/ossec-init.conf
+    chown root:wazuh /etc/ossec-init.conf
```
Comment (rauldpm) [inline arch/SPECS/wazuh-agent/arch/wazuh.install:58]: Use `${GROUP}` variable

File: arch/SPECS/wazuh-agent/arch/wazuh.install [hunk 5]
```diff
@@ -67,7 +67,7 @@ post_upgrade() {
 }
 
 post_install() {
-    if ! getent group | grep -q "^ossec" ; then
+    if ! getent group | grep -q "^wazuh" ; then
```
Comment (rauldpm) [inline arch/SPECS/wazuh-agent/arch/wazuh.install:70]: Please, change all references to wazuh user or group with the corresponding variables:

- `${USER}` 
- `${GROUP}`

**Review by rauldpm — CHANGES_REQUESTED** [review]
Please, review requested changes, thanks!

**Comment by rauldpm** [comment]
Also @avielw I am going to leave some comments that I cannot mark in the review itself:

- Is this parameter necessary? I don't see it being used, if so please remove it:

https://github.com/wazuh/wazuh-packages/blob/f79bec455fff232157f567bdd922dac8ad32ab62/arch/SPECS/wazuh-agent/arch/PKGBUILD#L6

- Change the conditional to `[[ "${use_local_specs}" == "no" ]]`

https://github.com/wazuh/wazuh-packages/blob/f79bec455fff232157f567bdd922dac8ad32ab62/arch/build.sh#L55

- Future packages are now identified by 99.99.0 instead of x.30.0, please, change related future generation

https://github.com/wazuh/wazuh-packages/blob/f79bec455fff232157f567bdd922dac8ad32ab62/arch/build.sh#L71

- Please, change the variables identified as $var to `"${var}"` throughout the code, for example:

https://github.com/wazuh/wazuh-packages/blob/f79bec455fff232157f567bdd922dac8ad32ab62/arch/build.sh#L69

Thanks for your contribution!

File: arch/SPECS/wazuh-agent/arch/wazuh.install (hunk 2 above)
Comment (avielw) [reply arch/SPECS/wazuh-agent/arch/wazuh.install:5]: There is no ossec user, Because this installation script didn't ever work on a release version with the ossec user (The Arch support was only added in version 4.3 when then user was already changed to wazuh. this is why this script never worked).

Comment (avielw) [reply arch/SPECS/wazuh-agent/arch/wazuh.install:5]: After the package is successfully built (which was not possible until this fix), this is the output you should get:
<details><summary>installation</summary>
loading packages...
resolving dependencies...
looking for conflicting packages...

Packages (1) wazuh-agent-4.3.4-1

Total Installed Size:  25.02 MiB

:: Proceed with installation? [Y/n] 
(1/1) checking keys in keyring                                  [###################################] 100%
(1/1) checking package integrity                                [###################################] 100%
(1/1) loading package files                                     [###################################] 100%
(1/1) checking for file conflicts                               [###################################] 100%
(1/1) checking available disk space                             [###################################] 100%
:: Processing package changes...
(1/1) installing wazuh-agent                                    [###################################] 100%
Creating group 'wazuh' with GID 958.
Creating user 'wazuh' (n/a) with UID 958 and GID 958.
:: Running post-transaction hooks...
(1/3) Creating system user accounts...
(2/3) Reloading system manager configuration...
(3/3) Arming ConditionNeedsUpdate...
</details>
<details><summary>uninstallation</summary>
checking dependencies...

Packages (1) wazuh-agent-4.3.4-1

Total Removed Size:  25.02 MiB

:: Do you want to remove these packages? [Y/n] 
:: Processing package changes...
(1/1) removing wazuh-agent                                      [###################################] 100%
warning: /var/ossec/etc/ossec.conf saved as /var/ossec/etc/ossec.conf.pacsave
:: Running post-transaction hooks...
(1/2) Reloading system manager configuration...
(2/2) Arming ConditionNeedsUpdate...
</details>
<details><summary>update</summary>
loading packages...
warning: wazuh-agent-4.3.4-1 is up to date -- reinstalling
resolving dependencies...
looking for conflicting packages...

Packages (1) wazuh-agent-4.3.4-1

Total Installed Size:  25.02 MiB
Net Upgrade Size:       0.00 MiB

:: Proceed with installation? [Y/n] 
(1/1) checking keys in keyring                                  [###################################] 100%
(1/1) checking package integrity                                [###################################] 100%
(1/1) loading package files                                     [###################################] 100%
(1/1) checking for file conflicts                               [###################################] 100%
(1/1) checking available disk space                             [###################################] 100%
:: Processing package changes...
(1/1) reinstalling wazuh-agent                                  [###################################] 100%
:: Running post-transaction hooks...
(1/3) Creating system user accounts...
(2/3) Reloading system manager configuration...
(3/3) Arming ConditionNeedsUpdate...
</details>

**Comment by avielw** [comment]
The url variable is not required but it is conventionally used in PKGBUILD files to give information of the origin of the sources:
https://wiki.archlinux.org/title/PKGBUILD#url

I've fixed what you've asked. I hope I didn't miss anything.

**Review by rauldpm — CHANGES_REQUESTED** [review]
In general, the changes made seem fine to me, but I make you a request for more changes, in particular, please review the package uninstall process, since if the package is uninstalled without stopping the services (using per example wazuh-control stop), the system is left with wazuh processes running (wazuh-execd, wazuh-agentd, wazuh-syscheckd...,  etc)

**Comment by avielw** [comment]
I think I've fixed what you've asked for except for the uninstall issues.

As far as I know, (and this is true for all the packages that installs any service in Arch), the convention in Arch is that the package only installs the files, and you need to enable/start/disable/stop the service manually (It's not that we cannot start/stop that service, but this is the convention).

This means that old processes that keeps running, and files that aren't deleted, because the service wasn't stopped before an uninstallation, is a behavior that happens for all the services you install in Arch, and not just wazuh.

I have seen the files that was not deleted, those are the files that wazuh created - configurations, logs, etc. (which also is the convention to not delete, in order to not delete important data while uninstalling a package), and in my case, if I've stopped the service before uninstalling wazuh, the /var/ossec/var directory was also deleted (which make sense because it contains pid files when the service is running).

So in summary, I think that stopping the services and removing some extra files, would be an unexpected behavior for an Arch machine.

**Comment by rauldpm** [comment]
> I think I've fixed what you've asked for except for the uninstall issues.
> 
> As far as I know, (and this is true for all the packages that installs any service in Arch), the convention in Arch is that the package only installs the files, and you need to enable/start/disable/stop the service manually (It's not that we cannot start/stop that service during the un/installation, but this is the convention).
> 
> This means that old processes that keeps running, and files that aren't deleted, because the service wasn't stopped before an uninstallation, is a behavior that happens for all the services you install in Arch, and not just wazuh.
> 
> I have seen the files that was not deleted, those are the files that wazuh created - configurations, logs, etc. (which also is the convention to not delete, in order to not delete important data while uninstalling a package), and in my case, if I've stopped the service before uninstalling wazuh, the /var/ossec/var directory was also deleted (which make sense because it contains pid files when the service is running).
> 
> So in summary, I think that stopping the services and removing some extra files, would be an unexpected behavior for an Arch machine.

Hello @avielw 

All the packages we provide follow the logic of stopping daemons that the same package raises so as not to leave orphaned daemons, since we cannot allow this case to occur, since it would break installations and, on some occasions, upgrades. We consider that it is a necessary security measure and although it is not a convention, our packages must carry it.

So we have two options:

1. Incorporate said change
2. We leave the PR in draft, we add a comment to the issue indicating that we have to make this change and we do it when we can according to our roadmap, which right now, i could not give you a completion date

Therefore, the changes that we ask to merge this PR is that the package stops its daemons when uninstalled, on the extra files, in principle, nothing should be done because, in fact, they are related to not stopping the daemons.

Regards, Raúl.

**Comment by avielw** [comment]
I think it's the wrong choice for Arch, but I'll add it anyway. Which files do you want to keep in case they've changed, when uninstalling? (for example the configuration files? the logs?)

File: arch/build.sh [hunk 6]
```diff
@@ -52,23 +45,23 @@ tmp_specs_path=${tmp_dir}/specs
...
 
-if [ "${use_local_specs}" = "no" ]; then
+if [[ "${use_local_specs}" == "no" ]]; then
     mkdir -p ${tmp_specs_path}
     cd ${tmp_specs_path}
     specs_path=${tmp_specs_path}
     curl -sL https://github.com/wazuh/wazuh-packages/tarball/${wazuh_packages_branch} | tar zx
     specs_path=`pwd`/$(find . -type d -name "SPECS" -path "*arch*")
-    tmp_specs_path=$specs_path
+    tmp_specs_path=${sepcs_path}
```
Comment (rauldpm) [inline arch/build.sh:54]: Testing the future package generation i found this typo, please fix it

**Review by rauldpm — CHANGES_REQUESTED** [review]
Please fix the typo

File: arch/SPECS/wazuh-agent/arch/wazuh.install [hunk 7]
```diff
@@ -80,3 +83,29 @@ post_install() {
 
     common
 }
+
+pre_remove() {
+    systemctl disable wazuh-agent.service 2>/dev/null || true
+    systemctl stop wazuh-agent.service 2>/dev/null || true
```
Comment (rauldpm) [inline arch/SPECS/wazuh-agent/arch/wazuh.install:89]: Could you use the wazuh-control binary to stop the service? An example would be like this https://github.com/wazuh/wazuh-packages/blob/f86b8425cb2327f219e1f83d5d363d149b2a52b4/debs/SPECS/wazuh-agent/debian/prerm#L23 where we check with systemctl, service and finally with wazuh-control

I just generated new packages with the branch of your repository and now I see that the stop of the daemons fails again, this is essential for the upgrade since it must stop them before doing it.

<details><summary>Remove daemons error</summary>

```
[root@archlinux vagrant]# pacman -U wazuh-agent-4.5.0-1-x86_64.pkg.tar.zst 
loading packages...
resolving dependencies...
looking for conflicting packages...

Packages (1) wazuh-agent-4.5.0-1

Total Installed Size:  26.10 MiB

:: Proceed with installation? [Y/n] Y
(1/1) checking keys in keyring                                 [##################################] 100%
(1/1) checking package integrity                               [##################################] 100%
(1/1) loading package files                                    [##################################] 100%
(1/1) checking for file conflicts                              [##################################] 100%
(1/1) checking available disk space                            [##################################] 100%
:: Processing package changes...
(1/1) installing wazuh-agent                                   [##################################] 100%
:: Running post-transaction hooks...
(1/3) Creating system user accounts...
(2/3) Reloading system manager configuration...
(3/3) Arming ConditionNeedsUpdate...
[root@archlinux vagrant]# nano /var/ossec/etc/ossec.conf 
[root@archlinux vagrant]# ps aux | grep wazuh
root        4148  0.0  0.2   4092  2432 pts/0    S+   14:23   0:00 grep wazuh
[root@archlinux vagrant]# /var/ossec/bin/wazuh-control start
Starting Wazuh v4.5.0...
Started wazuh-execd...
Started wazuh-agentd...
Started wazuh-syscheckd...
Started wazuh-logcollector...
Started wazuh-modulesd...
Completed.
[root@archlinux vagrant]# ps aux | grep wazuh
root        4177  0.0  0.2  24932  2744 ?        Sl   14:23   0:00 /var/ossec/bin/wazuh-execd
wazuh       4188  0.0  0.6 172448  6216 ?        Sl   14:23   0:00 /var/ossec/bin/wazuh-agentd
root        4199  0.0  0.5  99108  5604 ?        SNl  14:23   0:00 /var/ossec/bin/wazuh-syscheckd
root        4208  0.0  1.2 336064 12340 ?        Sl   14:23   0:00 /var/ossec/bin/wazuh-logcollector
root        4224  0.5  1.3 521200 13212 ?        Sl   14:23   0:00 /var/ossec/bin/wazuh-modulesd
root        4263  0.0  0.2   4092  2304 pts/0    S+   14:23   0:00 grep wazuh
[root@archlinux vagrant]# pacman -R wazuh-agent
checking dependencies...

Packages (1) wazuh-agent-4.5.0-1

Total Removed Size:  26.10 MiB

:: Do you want to remove these packages? [Y/n] Y
:: Processing package changes...
(1/1) removing wazuh-agent                                     [##################################] 100%
warning: /var/ossec/etc/ossec.conf saved as /var/ossec/etc/ossec.conf.pacsave
:: Running post-transaction hooks...
(1/2) Reloading system manager configuration...
(2/2) Arming ConditionNeedsUpdate...
[root@archlinux vagrant]# ps aux | grep wazuh
root        4177  0.0  0.2  24932  2744 ?        Sl   14:23   0:00 /var/ossec/bin/wazuh-execd
wazuh       4188  0.0  0.6 172448  6216 ?        Sl   14:23   0:00 /var/ossec/bin/wazuh-agentd
root        4199  0.0  0.5  99108  5604 ?        SNl  14:23   0:00 /var/ossec/bin/wazuh-syscheckd
root        4208  0.0  1.2 336064 12340 ?        Sl   14:23   0:00 /var/ossec/bin/wazuh-logcollector
root        4224  0.1  1.3 521200 13212 ?        Sl   14:23   0:00 /var/ossec/bin/wazuh-modulesd
root        4311  0.0  0.2   4092  2452 pts/0    S+   14:23   0:00 grep wazuh
[root@archlinux vagrant]#
```

</details>

Also i found that the deployment variables are not working and the package install show errors:

- https://documentation.wazuh.com/current/user-manual/deployment-variables/deployment-variables-linux.html

<details><summary>Deployment variables error</summary>

```
[root@archlinux vagrant]# WAZUH_MANAGER="192.168.57.102" pacman -U wazuh-agent-4.5.0-1-x86_64.pkg.tar.zst 
loading packages...
resolving dependencies...
looking for conflicting packages...

Packages (1) wazuh-agent-4.5.0-1

Total Installed Size:  26.10 MiB

:: Proceed with installation? [Y/n] Y
(1/1) checking keys in keyring                                 [##################################] 100%
(1/1) checking package integrity                               [##################################] 100%
(1/1) loading package files                                    [##################################] 100%
(1/1) checking for file conflicts                              [##################################] 100%
(1/1) checking available disk space                            [##################################] 100%
:: Processing package changes...
(1/1) installing wazuh-agent                                   [##################################] 100%
touch: cannot touch '/logs/ossec.log': No such file or directory
chmod: cannot access '/logs/ossec.log': No such file or directory
chown: cannot access '/logs/ossec.log': No such file or directory
grep: /etc/ossec.conf: No such file or directory
grep: /etc/ossec.conf: No such file or directory
sed: can't read /etc/ossec.conf: No such file or directory
/var/ossec/packages_files/agent_installation_scripts/src/init/register_configure_agent.sh: line 58: /logs/ossec.log: No such file or directory
:: Running post-transaction hooks...
(1/3) Creating system user accounts...
(2/3) Reloading system manager configuration...
(3/3) Arming ConditionNeedsUpdate...
[root@archlinux vagrant]# grep "MANAGER_IP" /var/ossec/etc/ossec.conf
      <address>MANAGER_IP</address>
[root@archlinux vagrant]#
```

</details>

Some examples on how to manage these variables can be found in the SPECS of deb and rpm packages generation process

https://github.com/wazuh/wazuh-packages/blob/f86b8425cb2327f219e1f83d5d363d149b2a52b4/debs/SPECS/wazuh-agent/debian/postinst#L125
https://github.com/wazuh/wazuh-packages/blob/f86b8425cb2327f219e1f83d5d363d149b2a52b4/rpms/SPECS/wazuh-agent.spec#L253

**Review by rauldpm — CHANGES_REQUESTED** [review]
Future package generation seems ok to me, but review the uninstall package process and deployment variables please

**Review by rauldpm — APPROVED** [review]
Tests:

- Package installation OK
- Package removal OK
- Daemons stops when removal OK
- Alerts generated OK
- Deployment variables OK (tested with WAZUH_MANAGER and WAZUH_PROTOCOL)
- Future package generation OK
- Package upgrade OK (to future)
  - Daemons pid change after upgrade
  - Alerts received after upgrade
  - Agent connection persists after upgrade 


<details>Agent install, upgrade, daemons stop, uninstall and deployment variables</summary>

```
[root@archlinux vagrant]# WAZUH_MANAGER="192.168.57.102" pacman -U wazuh-agent-4.5.0-1-x86_64.pkg.tar.zst 
loading packages...
resolving dependencies...
looking for conflicting packages...

Packages (1) wazuh-agent-4.5.0-1

Total Installed Size:  26.10 MiB

:: Proceed with installation? [Y/n] Y
(1/1) checking keys in keyring                                 [##################################] 100%
(1/1) checking package integrity                               [##################################] 100%
(1/1) loading package files                                    [##################################] 100%
ç(1/1) checking for file conflicts                              [##################################] 100%
(1/1) checking available disk space                            [##################################] 100%
:: Processing package changes...
(1/1) installing wazuh-agent                                   [##################################] 100%
:: Running post-transaction hooks...
(1/3) Creating system user accounts...
(2/3) Reloading system manager configuration...
(3/3) Arming ConditionNeedsUpdate...
[root@archlinux vagrant]# grep "address" /var/ossec/etc/ossec.conf 
      <address>192.168.57.102</address>
[root@archlinux vagrant]# /var/ossec/bin/wazuh-control start
Starting Wazuh v4.5.0...
Started wazuh-execd...
Started wazuh-agentd...
Started wazuh-syscheckd...
Started wazuh-logcollector...
Started wazuh-modulesd...
Completed.
[root@archlinux vagrant]# ps aux | grep wazuh
root        7562  0.0  0.2  24740  2712 ?        Sl   16:38   0:00 /var/ossec/bin/wazuh-execd
wazuh       7571  0.0  0.8 172452  7976 ?        Sl   16:38   0:00 /var/ossec/bin/wazuh-agentd
root        7582  0.0  0.3  99064  3876 ?        SNl  16:38   0:00 /var/ossec/bin/wazuh-syscheckd
root        7591  0.0  0.4 336068  4240 ?        Sl   16:38   0:00 /var/ossec/bin/wazuh-logcollector
root        7607  0.3  1.5 521204 15808 ?        Sl   16:38   0:00 /var/ossec/bin/wazuh-modulesd
root        7648  0.0  0.2   4092  2396 pts/0    S+   16:38   0:00 grep wazuh
[root@archlinux vagrant]# touch /home/vagrant/test_preupdate
[root@archlinux vagrant]# pacman -U wazuh-agent-99.99.0-1-x86_64.pkg.tar.zst 
loading packages...
resolving dependencies...
looking for conflicting packages...

Packages (1) wazuh-agent-99.99.0-1

Total Installed Size:  26.10 MiB
Net Upgrade Size:       0.00 MiB

:: Proceed with installation? [Y/n] Y
(1/1) checking keys in keyring                                 [##################################] 100%
(1/1) checking package integrity                               [##################################] 100%
(1/1) loading package files                                    [##################################] 100%
(1/1) checking for file conflicts                              [##################################] 100%
(1/1) checking available disk space                            [##################################] 100%
:: Processing package changes...
(1/1) upgrading wazuh-agent                                    [##################################] 100%
:: Running post-transaction hooks...
(1/3) Creating system user accounts...
(2/3) Reloading system manager configuration...
(3/3) Arming ConditionNeedsUpdate...
[root@archlinux vagrant]# /var/ossec/bin/wazuh-control status
wazuh-modulesd is running...
wazuh-logcollector is running...
wazuh-syscheckd is running...
wazuh-agentd is running...
wazuh-execd is running...
[root@archlinux vagrant]# ps aux | grep wazuh
root        8494  0.0  0.2  24740  2676 ?        Sl   16:42   0:00 /var/ossec/bin/wazuh-execd
wazuh       8503  0.4  0.9 246576  9780 ?        Sl   16:42   0:00 /var/ossec/bin/wazuh-agentd
root        8516 12.0  1.0 321204 10144 ?        SNl  16:42   0:04 /var/ossec/bin/wazuh-syscheckd
root        8527  0.0  0.4 336076  4612 ?        Sl   16:42   0:00 /var/ossec/bin/wazuh-logcollector
root        8549  0.7  1.3 652276 13892 ?        Sl   16:42   0:00 /var/ossec/bin/wazuh-modulesd
root        9000  0.0  0.2   4092  2436 pts/0    S+   16:43   0:00 grep wazuh
[root@archlinux vagrant]# touch /home/vagrant/test_postupgrade
[root@archlinux vagrant]# ps aux | grep wazuh
root        9129  0.0  0.2   4092  2388 pts/0    S+   16:48   0:00 grep wazuh
[root@archlinux vagrant]# 
```

```
[root@archlinux vagrant]# WAZUH_MANAGER="192.168.57.102" WAZUH_PROTOCOL="udp" pacman -U wazuh-agent-4.5.0-1-x86_64.pkg.tar.zst 
loading packages...
resolving dependencies...
looking for conflicting packages...

Packages (1) wazuh-agent-4.5.0-1

Total Installed Size:  26.10 MiB

:: Proceed with installation? [Y/n] Y
(1/1) checking keys in keyring                                                                                                 [#############################################################################] 100%
(1/1) checking package integrity                                                                                               [#############################################################################] 100%
(1/1) loading package files                                                                                                    [#############################################################################] 100%
(1/1) checking for file conflicts                                                                                              [#############################################################################] 100%
(1/1) checking available disk space                                                                                            [#############################################################################] 100%
:: Processing package changes...
(1/1) installing wazuh-agent                                                                                                   [#############################################################################] 100%
:: Running post-transaction hooks...
(1/3) Creating system user accounts...
(2/3) Reloading system manager configuration...
(3/3) Arming ConditionNeedsUpdate...
[root@archlinux vagrant]# grep "address" /var/ossec/etc/ossec.conf 
      <address>192.168.57.102</address>
[root@archlinux vagrant]# grep "udp" /var/ossec/etc/ossec.conf 
      <protocol>udp</protocol>
[root@archlinux vagrant]#
```

</details>

<details><summary>Generated alert before upgrade</summary>

```
** Alert 1656002402.785337: - ossec,syscheck,syscheck_entry_added,syscheck_file,pci_dss_11.5,gpg13_4.11,gdpr_II_5.1.f,hipaa_164.312.c.1,hipaa_164.312.c.2,nist_800_53_SI.7,tsc_PI1.4,tsc_PI1.5,tsc_CC6.1,tsc_CC6.8,tsc_CC7.2,tsc_CC7.3,
2022 Jun 23 16:40:02 (archlinux) any->syscheck
Rule: 554 (level 5) -> 'File added to the system.'
File '/home/vagrant/test_preupdate' added
Mode: realtime

Attributes:
 - Size: 0
 - Permissions: rw-r--r--
 - Date: Thu Jun 23 16:40:02 2022
 - Inode: 41939
 - User: root (0)
 - Group: root (0)
 - MD5: d41d8cd98f00b204e9800998ecf8427e
 - SHA1: da39a3ee5e6b4b0d3255bfef95601890afd80709
 - SHA256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
```

</details>

<details><summary>Stop and start alerts when upgrade</summary>

```
** Alert 1656002575.788247: - ossec,pci_dss_10.6.1,pci_dss_10.2.6,gpg13_10.1,gdpr_IV_35.7.d,hipaa_164.312.b,nist_800_53_AU.6,nist_800_53_AU.14,nist_800_53_AU.5,tsc_CC7.2,tsc_CC7.3,tsc_CC6.8,
2022 Jun 23 16:42:55 (archlinux) any->wazuh-remoted
Rule: 506 (level 3) -> 'Ossec agent stopped.'
ossec: Agent stopped: 'archlinux->any'.

** Alert 1656002576.788577: - ossec,pci_dss_10.6.1,pci_dss_10.2.6,gpg13_10.1,gdpr_IV_35.7.d,hipaa_164.312.b,nist_800_53_AU.6,nist_800_53_AU.14,nist_800_53_AU.5,tsc_CC7.2,tsc_CC7.3,tsc_CC6.8,
2022 Jun 23 16:42:56 (archlinux) any->ossec
Rule: 503 (level 3) -> 'Ossec agent started.'
ossec: Agent started: 'archlinux->any'.
```

</details>

<details><summary>Generated alert after upgrade</summary>

```
** Alert 1656002640.790455: - ossec,syscheck,syscheck_entry_added,syscheck_file,pci_dss_11.5,gpg13_4.11,gdpr_II_5.1.f,hipaa_164.312.c.1,hipaa_164.312.c.2,nist_800_53_SI.7,tsc_PI1.4,tsc_PI1.5,tsc_CC6.1,tsc_CC6.8,tsc_CC7.2,tsc_CC7.3,
2022 Jun 23 16:44:00 (archlinux) any->syscheck
Rule: 554 (level 5) -> 'File added to the system.'
File '/home/vagrant/test_postupgrade' added
Mode: realtime

Attributes:
 - Size: 0
 - Permissions: rw-r--r--
 - Date: Thu Jun 23 16:44:00 2022
 - Inode: 42258
 - User: root (0)
 - Group: root (0)
 - MD5: d41d8cd98f00b204e9800998ecf8427e
 - SHA1: da39a3ee5e6b4b0d3255bfef95601890afd80709
 - SHA256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
```

</details>

**Review by okynos — APPROVED** [review]

**Comment by okynos** [comment]
Thanks @avielw for your contribution.
